tests/integration/components/spread-test.js

```diff
@@ -1,30 +1,235 @@
+
+        expect($hook('spread-property').text().trim()).to.equal('Crazy')
+      })
+
+      it('should bind new source functions as local functions', function () {
+        this.set('options.onClick', handler)
+
+        $hook('spread-test').click()
+
+        expect(handler).to.have.callCount(1)
+      })
+
+      it('should update new local properties when the source property changes', function () {
+        this.set('options.property', 'Crazy')
+
+        expect($hook('spread-property').text().trim()).to.equal('Crazy')
+
+        this.set('options.property', 'Booya')
+
+        expect($hook('spread-property').text().trim()).to.equal('Booya')
+      })
+    })
+
+    describe('and using a custom spread property', function () {
+      beforeEach(function () {
+        this.set('options', Ember.Object.create({}))
+
+        this.render(hbs`
+          {{spread-test
+            foo=options
+            spreadOptions=(hash
+              property='foo'
+              source=(hash
+                object=this
+                property='options'
+              )
+            )
+          }}
+        `)
+      })
+
+      it('should bind new source properties as local properties', function () {
+        this.set('options.property', 'Crazy')
+
+        expect($hook('spread-property').text().trim()).to.equal('Crazy')
+      })
+
+      it('should bind new source functions as local functions', function () {
+        this.set('options.onClick', handler)
+
+        $hook('spread-test').click()
+
+        expect(handler).to.have.callCount(1)
+      })
+
+      it('should update new local properties when the source property changes', function () {
+        this.set('options.property', 'Crazy')
+
+        expect($hook('spread-property').text().trim()).to.equal('Crazy')
+
+        this.set('options.property', 'Booya')
+
+        expect($hook('spread-property').text().trim()).to.equal('Booya')
+      })
+    })
+
+    describe('and the component is destroyed', function () {
+      beforeEach(function () {
+        this.setProperties({
+          options: Ember.Object.create({}),
+          condition: true
+        })
+
+        this.render(hbs`
+          {{spread-test
+            options=options
+            spreadOptions=(hash
+              source=(hash
+                object=this
+                property='options'
+              )
+            )
+          }}
+
+          {{#if condition}}
+            {{spread-test
+              options=options
+              spreadOptions=(hash
+                source=(hash
+                  object=this
+                  property='options'
+                )
+              )
+            }}
+          {{/if}}
+        `)
+      })
+
+      it('should clean up the source property listener', function () {
+        expect(this.get('options._spreadListeners').length).to.equal(2)
+
+        this.set('condition', false)
+
+        expect(this.get('options._spreadListeners').length).to.equal(1)
```

Comment:
`chai` actually has a [`lengthOf()`](http://chaijs.com/api/bdd/#method_lengthof) helper so you could do
```js
expect(this.get('options._spreadListeners')).to.have.lengthOf(1)
```
If you want to.